hadoop-hdds/common/src/test/java/org/apache/hadoop/hdds/utils/db/cache/TestTableCacheImpl.java

```diff
@@ -88,6 +89,55 @@ public void testPartialTableCache() {
   }
 
 
+  @Test
+  public void testPartialTableCacheWithNotContinousEntries() throws Exception {
+    int totalCount = 0;
+    int insertedCount = 3000;
+    for (long i=0; i<insertedCount; i+=2) {
+      tableCache.put(new CacheKey<>(Long.toString(i)),
+          new CacheValue<>(Optional.of(Long.toString(i)), i));
+      totalCount++;
+    }
+
+    Assert.assertEquals(totalCount, tableCache.size());
+
+    ArrayList<Long> epochs = new ArrayList();
+    epochs.add(0L);
+    epochs.add(10L);
+    epochs.add(300L);
+    epochs.add(500L);
+    epochs.add(1000L);
+
+    tableCache.cleanup(epochs);
+
+    final int count = totalCount;
+
+    // If cleanup policy is manual entries should have been removed.
+    if (cacheCleanupPolicy == TableCacheImpl.CacheCleanupPolicy.MANUAL) {
+      GenericTestUtils.waitFor(() -> {
+        return count - epochs.size() == tableCache.size();
+      }, 100, 10000);
+
+      // Check remaining entries exist or not and deleted entries does not
+      // exist.
+      for (long i = 0; i < insertedCount; i += 2) {
+        if (!epochs.contains(i)) {
```

Comment:
same here, not sure why we are incrementing by +2.

Reply:
Just adding entries in the cache which are not in continuous order.

Reply:
I am not sure 0,2,4,6 satisfies the non-continuous order property you are seeking here.

Reply:
Updated test to use according to my original intention of i+=2